lxd/device/nic_routed.go

```diff
@@ -239,6 +239,34 @@ func (d *nicRouted) Start() (*deviceConfig.RunConfig, error) {
 		}
 	}
 
+	// Ensure the configured IP addresses are not in use.
+	for _, keyPrefix := range []string{"ipv4", "ipv6"} {
+		addr, ok := d.config[fmt.Sprintf("%s.address", keyPrefix)]
+		if ok {
+			p, err := ping.NewPinger(addr)
```

Comment:
Looking at https://github.com/go-ping/ping this doesn't appear to be using an ARP or NDP neighbour discovery, but instead ICMP ping. This isn't what we're after here, as the remote machine may have a firewall and wouldn't be responding to ICMP pings, but would respond to ARP/NDP.

Reply:
Ah I see, will revert to the arping library and find something suitable for NDP.